Suggested edits for identity doc updates (#29339)
hashicorp/vault: website/content/docs/upgrading/deduplication/different-case.mdx

[WARN]  identity: entity-alias "alias-cAsE" with mount accessor "auth_userpass_34aca7ec" duplicates 1 others: id=2992253b-1e99-4e47-b6f9-afb0c7cedf7a canonical_id=7da76b0d-fe9b-a125-3362-2a8ff055dcf8 force_deduplication="would merge others into this entity"
[WARN]  identity: entity-alias "alias-case" with mount accessor "auth_userpass_a555989a" duplicates 1 others: id=0e4bd46e-a868-4dd4-a34a-cb73f097e3a5 canonical_id=1ce06951-e2fd-0923-8ae0-a0e2c6c2378b force_deduplication="would merge into entity 37f41fcf-5e15-f13f-248d-9fa8405b1ecd"
[WARN]  identity: entity-alias "alias-cAsE" with mount accessor "auth_userpass_a555989a" duplicates 1 others: id=6cafe546-665c-4bd0-a4fd-5e699ba413c1 canonical_id=37f41fcf-5e15-f13f-248d-9fa8405b1ecd force_deduplication="would merge others into this entity"
[WARN]  identity: end of different-case entity-alias duplicates
```

The duplicate alias in the example log are distinct, despite have the same name,
because they come from different authN mounts. When resolving duplicates,
consider the set of duplicate names from each authN mount separately.

Before resolving different-case duplicates, you must determine if the case

> difference is a **true duplicate** or **false positive**:

Reviewer: We talked about this and thought `mergable` vs `unmergable` is a good way to communicate these.

**You must review different-case entity alias duplicates before enabling forced
identity deduplication due to the potential security risk of incorrectly merging
distinct aliases**.

<Tip title="Assumptions">

- You have [identified the duplicate entity aliases](/vault/docs/upgrading/identity-deduplication). 
- You have admin permission on the relevant Vault server or cluster.

</Tip>

## Why entity alias duplicates happen

Historical bugs in Vault allowed authN mounts to create duplicate entries of the
same entity alias if the aliases used difference cases. For example, if an
external LDAP system returns unnormalized usernames or changes the normalization
scheme over time. By default, Vault uses case-insensitive matching for the
usernames and considers `username` and `uSeRnAmE` to be the same alias. But,
past bugs have left some users with different-case duplicates in storage. To
avoid the security impact of incorrectly merging identities that do not
represent the same user, Vault tries to honor past behavior by switching the
identity engine to use case-sensitive matching mode.
But case-sensitive matching 
This mode is not well supported however  

> as it deviates from Vault's typical assumptions. Remaining in this mode can lead to unexpected behavior. 

Reviewer: I think this summary which was part of my above suggestion ended up broken in the edit somewhere. I think the wording is good once the partial sentence is removed and this is merged with the final sentence of the paragraph above.

[WARN]  identity: entity-alias "alias-case" with mount accessor "auth_userpass_a555989a" duplicates 1 others: id=0e4bd46e-a868-4dd4-a34a-cb73f097e3a5 canonical_id=1ce06951-e2fd-0923-8ae0-a0e2c6c2378b force_deduplication="would merge into entity 37f41fcf-5e15-f13f-248d-9fa8405b1ecd"
[WARN]  identity: entity-alias "alias-cAsE" with mount accessor "auth_userpass_a555989a" duplicates 1 others: id=6cafe546-665c-4bd0-a4fd-5e699ba413c1 canonical_id=37f41fcf-5e15-f13f-248d-9fa8405b1ecd force_deduplication="would merge others into this entity"
[WARN]  identity: end of different-case entity-alias duplicates
```

The duplicate alias in the example log are distinct, despite have the same name,
because they come from different authN mounts. When resolving duplicates,
consider the set of duplicate names from each authN mount separately.

Before resolving different-case duplicates, you must determine if the case
difference is a **mergeable duplicate** or **unmergeable duplicate**:

> - For mergeable duplicates, the case difference is **unintentional** and the
>   duplicate entries represent the same logical entity. You should deduplicate
>   true duplicates to reflect the expected default Vault behavior.

Reviewer: ```suggestion
- For mergeable duplicates, the case difference is **unintentional** and the
  duplicate entries represent the same logical entity. It is safe to ignore these 
  and allow them to be merged automatically when the `force-identity-deduplication` 
  flag is activated.
```
Unsure if this is needed here as we say it below, but it feels strange to me to say "you should deduplicate" when the instructions to fix below are "ignore these". The instruction here sounds like "action required" when it's really "no action required"?

> - For unmergeable duplicates, the case difference is **deliberate** and the
>   source identity provider **intended** to distinguish between different
>   entities using case. For example, `Alice` and `alice` are different entities
>   that correspond to different humans with distinct permission sets.

Reviewer: ```suggestion
- For unmergeable duplicates, the case difference is **deliberate** and the
  source identity provider **intended** to distinguish between different
  entities using case. For example, `Alice` and `alice` are different entities
  that correspond to different humans with distinct permission sets. **These duplicates must be resolved before activating the deduplication flag.**
```

Success! Data deleted (if it existed) at: identity/entity-alias/id/df3568a4-3b65-4104-9481-1129ecbed72f
```

</CodeBlockConfig>

</Tab>

<Tab heading="API" group="api">

Call the [`/identity/entity-alias/id/{id}`](/vault/api-docs/secret/identity/entity-alias#delete-entity-alias-by-id)
endpoint to delete the duplicate entry:

```shell-session
$ curl                                      \
   --request DELETE                         \
   --header "X-Vault-Token: ${VAULT_TOKEN}" \
   ${VAULT_ADDR}/v1/identity/entity-alias/id/<id>
```

For example, if one of the duplicate entity log entries includes
`id=df3568a4-3b65-4104-9481-1129ecbed72f`:

<CodeBlockConfig hideClipboard="true">

```shell-session
$ curl                                      \
   --request DELETE                         \
   --header "X-Vault-Token: ${VAULT_TOKEN}" \
   ${VAULT_ADDR}/v1/identity/entity-alias/id/df3568a4-3b65-4104-9481-1129ecbed72f | jq
```

</CodeBlockConfig>

</Tab>

</Tabs>

After you resolve the duplicates by deleting them or reconfiguring the relevant
services, recheck the system logs to confirm the duplicates no longer appear in

> the log during unseal.

Reviewer: ```suggestion
the log during the next unseal.
```
Not sure if we should spell out - you might need to explicitly restart a standby node to observe this or wait for it to happen naturally?
